dali/base/dacsds.cpp

```diff
@@ -2236,32 +2239,58 @@ bool CClientSDSManager::updateEnvironment(IPropertyTree *newEnv, bool forceGroup
 
 //////////////
 
+static ISDSManager * activeSDSManager=NULL;
+static ISDSManager * savedSDSManager=NULL;
+
+MODULE_INIT(INIT_PRIORITY_STANDARD)
+{
+    return true;
+}
+MODULE_EXIT()
+{
+    delete activeSDSManager;
+    activeSDSManager = nullptr;
+    delete savedSDSManager;
+    savedSDSManager = nullptr;
+}
+
 ISDSManager &querySDS()
 {
     CriticalBlock block(SDScrit);
-    if (SDSManager)
-        return *SDSManager;
+    if (activeSDSManager)
+        return *activeSDSManager;
     else if (!queryCoven().inCoven())
     {
-        if (!SDSManager)
-            SDSManager = new CClientSDSManager();
+        if (!activeSDSManager)
+            activeSDSManager = new CClientSDSManager();
 
-        return *SDSManager;
+        return *activeSDSManager;
     }
     else
     {
-        SDSManager = &querySDSServer();
-        return *SDSManager;
+        activeSDSManager = &querySDSServer();
+        return *activeSDSManager;
     }
 }
 
 void closeSDS()
 {
     CriticalBlock block(SDScrit);
-    if (SDSManager) {
+
+    //In roxie this is called when connection to dali is lost, but other threads can still be processing
+    //CRemoteConnections (see HPCC-32410), which use an ISDSManager member - accessing a stale manager.
+    //There can be similar issues at closedown if threads have not been cleaned up properly.
+    //Do not delete the active SDS manager immediately - save it so that it is deleted on the next call/closedown.
+    ISDSManager * toDelete = savedSDSManager;
+    savedSDSManager = activeSDSManager;
+    activeSDSManager = nullptr;
+    if (savedSDSManager || toDelete)
+    {
```

Comment:
trivial: inconsistent block bracing. I normally make all within the method Allman style like this when adding code that introduces new ones.